src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -506,6 +508,61 @@ long IslandWindow::_calculateTotalSize(const bool isWidth, const long clientSize
     case WM_THEMECHANGED:
         UpdateWindowIconForActiveMetrics(_window.get());
         return 0;
+    case WM_WINDOWPOSCHANGING:
+    {
+        // GH#10274 - if the quake window gets moved to another monitor via aero
+        // snap (win+shift+arrows), then re-adjust the size for the new monitor.
+        if (IsQuakeWindow())

```

Comment:
BTW Would it be possible to change `TerminalPage::WindowName` to return a `const hstring&` (as it's used inside `IsQuakeWindow()`)?

Reply:
Two reasons not to-

1. HSTRINGs are reference-counted, not copied. If `WindowName` is not `hstring`, it should be.
2. Can't return references across WinRT ABI boundaries

Reply:
Oh I didn't consider whether this might be a COM API.